static/js/subs.js

```diff
@@ -345,10 +345,13 @@ exports.update_add_subscriptions_elements = function (allow_user_to_add_subs) {
 };
 
 exports.update_settings_for_unsubscribed = function (sub) {
-    stream_edit.hide_sub_settings(sub);
     exports.rerender_subscriptions_settings(sub);
     stream_ui_updates.update_check_button_for_sub(sub);
     stream_ui_updates.update_settings_button_for_sub(sub);
+    stream_ui_updates.update_regular_sub_settings(sub);
+    if (!sub.can_change_stream_permissions) {
+        $(".change-stream-privacy").hide();
+    }
```

Comment:
Can you split the extraction of the `!sub.can_change_stream_permissions` block into its own commit?  It'd make it easier to see that the other change don't do anything weird.